gui/src/callbacks.cc

```diff
@@ -438,6 +441,44 @@ void find_or_create_scroom(PresentationInterface::Ptr presentation)
   create_scroom(presentation);
 }
 
+void
+onDragDataReceived(GtkWidget *, GdkDragContext *, int, int,
+                        GtkSelectionData *seldata, guint, guint,
+                        gpointer)
+{
+  printf("Dropping file(s) onto SCROOM %s\n", (gchar*)seldata->data);
+  std::string urilist((gchar*)seldata->data);
+  
+  // Split all the filenames in the dropped uri list to a vector of strings
+  std::vector<std::string> dropped_filenames;
+  boost::split(dropped_filenames, urilist, boost::is_space());
+  
+  for(const auto & filename : dropped_filenames) {
+  	// check that the filename starts with file:///
+	if(filename.empty() || filename.rfind("file:///", 0) == std::string::npos) continue;
+	
+	try {
+	  load(filename.substr(8));
+	} 
+	catch (std::invalid_argument & ex) 
+	{
+	  boost::format warningFormat =
+          boost::format("Warning: unable to load file %s") % filename;
+      printf("%s\n", warningFormat.str().c_str());
+      if (gdk_display_get_default()) {
+        // We're not running headless, don't open the popup
+        // We don't have a pointer to the parent window, so nullptr should suffice
+        GtkWidget *dialog =
+          gtk_message_dialog_new(nullptr, GTK_DIALOG_DESTROY_WITH_PARENT, GTK_MESSAGE_WARNING,
+        						 GTK_BUTTONS_CLOSE, warningFormat.str().c_str());
```

Comment:
As a rule, I'm not a big fan of modal dialogs, since they block scroom, even in the case where there might be other work to do (such as loading other files). Thus, they violate the "as fast as possible" property I'd like scroom to have.

However, I agree that error reporting needs to improve, so I guess for now they have to do.